internal/stage_hp1.go

```diff
@@ -0,0 +1,101 @@
+// Stage HP2: History Write Test
+// This test checks if the shell supports writing command history to a file using the history -w command.
+// It sends a few commands, writes them to history, and verifies the history file contents.
+
+package internal
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/codecrafters-io/shell-tester/internal/logged_shell_asserter"
+	"github.com/codecrafters-io/shell-tester/internal/shell_executable"
+	"github.com/codecrafters-io/shell-tester/internal/test_cases"
+	"github.com/codecrafters-io/tester-utils/random"
+	"github.com/codecrafters-io/tester-utils/test_case_harness"
+)
+
+func testHP1(stageHarness *test_case_harness.TestCaseHarness) error {
+	logger := stageHarness.Logger
+	shell := shell_executable.NewShellExecutable(stageHarness)
+	asserter := logged_shell_asserter.NewLoggedShellAsserter(shell)
+
+	// Step 1: Create a temporary history file with some commands
+	historyFile := filepath.Join(os.TempDir(), random.RandomWord()+"_shell_history_test")
+	defer os.Remove(historyFile)
+
+	// Set HISTFILE to /dev/null before starting the shell
+	shell.Setenv("HISTFILE", "/dev/null")
+
+	if err := asserter.StartShellAndAssertPrompt(true); err != nil {
+		return err
+	}
+
+	// Step 1: Create history file with random commands
+	nFileCommands := random.RandomInt(2, 5)
+	fileCommands := make([]string, nFileCommands)
+	for i := 0; i < nFileCommands; i++ {
+		fileCommands[i] = "echo " + strings.Join(random.RandomWords(random.RandomInt(2, 4)), " ")
+	}
+	historyFileContent := strings.Join(fileCommands, "\n") + "\n"
+	if err := os.WriteFile(historyFile, []byte(historyFileContent), 0644); err != nil {
+		logger.Errorf("Failed to write history file: %v", err)
+		return err
+	}
+
+	// print the history file content
+	content, err := os.ReadFile(historyFile)
+	if err != nil {
+		logger.Errorf("Failed to read history file: %v", err)
+		return err
+	}
+	logger.Debugf("History file content: \n%s", string(content))
```

Comment:
@Arpan-206 let's change the formatting of how this file is printed. Look at the interpreter challenge and see what we do these (it uses `secondaryLogger` or something like that, key thing is that the text should be white and not blue)